daemon/src/db.rs

```diff
@@ -493,6 +513,19 @@ pub async fn load_cfds_by_oracle_event_id(
     Ok(cfds)
 }
 
+fn convert_to_system_time(row_secs: i64, row_nanos: i32) -> SystemTime {
+    let secs = row_secs as i128;
+    let nanos = row_nanos as i128;
+    let offset_dt =
+        OffsetDateTime::from_unix_timestamp_nanos(secs * 1_000_000_000 + nanos).unwrap();
+
+    SystemTime::from(offset_dt)
+}
+
+fn convert_to_duration(row_secs: i64, row_nanos: i32) -> Duration {
+    Duration::new(row_secs, row_nanos)
+}
```

Comment:
Don't think this function is providing much benefit?

Reply:
Benefit is visual only; if nanoseconds are dropped then even that benefit is gone. :-)